### PR TITLE
Adds A Per-Slot Virginity Toggle

### PR DESCRIPTION
**Pull request by Kevak**
## About The Pull Request

I have seen multiple players ask for a way to toggle being a virgin. This enables such. The toggle is per-character and is overridden by the normal roles that override virginity. Namely duke/duchess _must_ be non-virgins as they popped out the heirs, archivist/steward are default virgins according to the code, and a good few church/clergy roles are default virgin.

## Why It's Good For The Game

More customization good. Also multiple players want it.

## Proof Of Testing

Screenshot 2024-12-11 034707
Screenshot 2024-12-11 035145

### CODE REVIEW

File: code/modules/client/preferences_savefile.dm [hunk 1]
```diff
@@ -355,6 +355,7 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 	S["feature_mcolor2"]					>> features["mcolor2"]
 	S["feature_mcolor3"]					>> features["mcolor3"]
 	S["feature_ethcolor"]					>> features["ethcolor"]
+	S["virginity"]			>> virginity
```
Comment (Heheniy) [inline code/modules/client/preferences_savefile.dm:358]:
```suggestion
	S["virginity"]							>> virginity
```

File: code/modules/client/preferences_savefile.dm [hunk 2]
```diff
@@ -475,6 +476,8 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 	body_markings = SANITIZE_LIST(body_markings)
 	validate_body_markings()
 
+	virginity	= sanitize_integer(virginity, FALSE, TRUE, FALSE)
```
Comment (Heheniy) [inline code/modules/client/preferences_savefile.dm:479]:
```suggestion
	virginity = sanitize_integer(virginity, FALSE, TRUE, FALSE)
```

File: code/modules/jobs/job_types/roguetown/church/monk.dm [hunk 3]
```diff
@@ -94,3 +93,7 @@
 	var/datum/devotion/C = new /datum/devotion(H, H.patron)
 	C.grant_spells(H)
 	H.verbs += list(/mob/living/carbon/human/proc/devotionreport, /mob/living/carbon/human/proc/clericpray)
+
+/datum/outfit/job/roguetown/monk/pre_equip(mob/living/carbon/human/H)
```
Comment (Heheniy) [inline code/modules/jobs/job_types/roguetown/church/monk.dm:97]:
```suggestion
/datum/outfit/job/roguetown/monk/post_equip(mob/living/carbon/human/H)
```

File: code/modules/jobs/job_types/roguetown/church/priest.dm [hunk 4]
```diff
@@ -73,6 +72,10 @@
 //		H.underwear_color = CLOTHING_BLACK
 //		H.update_body()
 
+/datum/outfit/job/roguetown/priest/pre_equip(mob/living/carbon/human/H)
```
Comment (Heheniy) [inline code/modules/jobs/job_types/roguetown/church/priest.dm:75]:
```suggestion
/datum/outfit/job/roguetown/priest/post_equip(mob/living/carbon/human/H)
```

File: code/modules/jobs/job_types/roguetown/nobility/lady.dm [hunk 5]
```diff
@@ -66,6 +66,10 @@
 		H.change_stat("perception", 2)
 		H.change_stat("fortune", 5)
 
+/datum/outfit/job/roguetown/lady/pre_equip(mob/living/carbon/human/H)
```
Comment (Heheniy) [inline code/modules/jobs/job_types/roguetown/nobility/lady.dm:69]:
```suggestion
/datum/outfit/job/roguetown/lady/post_equip(mob/living/carbon/human/H)
```

File: code/modules/jobs/job_types/roguetown/church/monk.dm (hunk 3 above)
Comment (Kevak) [reply code/modules/jobs/job_types/roguetown/church/monk.dm:97]: Oh shit did I miss one? Oop

**Comment by notafurryy** [comment]
Aren't those roles supposed to be virgins for mechanical reasons? Cultists and pretty sure vamps have to kidnap them for some rituals, this could make things harder or easier for them.

**Comment by Kevak** [comment]
> Aren't those roles supposed to be virgins for mechanical reasons? Cultists and pretty sure vamps have to kidnap them for some rituals, this could make things harder or easier for them.

The roles that are by default virgin remain virgin. The toggle does not override them. It merely moves the application of virginity to the post_equip proc because pre_equip is called before prefs, and post_equip is called after. The rng virginity is the main thing that is liable to be impacted. (All non-default virgin roles had a chance to be a virgin round start. This simply lets players pick to have that a static thing.)

**Comment by Kevak** [comment]
Also @Heheniy I appreciate you catching those, kitty high and eepy. Thanks

**Pull request merged**